doc/api/packages.md

```diff
@@ -819,10 +747,216 @@ conditional exports for consumers could be to add an export, e.g.
 }
 ```
 
-[Conditional exports]: #packages_conditional_exports
+## Node.js `package.json` field definitions
+
+This section describes the fields used by the Node.js runtime. Other tools (such
+as [npm](https://docs.npmjs.com/creating-a-package-json-file)) may use
+additional fields which are ignored by Node.js and not documented here.
+
+### `"name"`
+<!-- YAML
+added:
+  - v12.16.0
+  - v13.1.0
+changes:
+  - version:
+    - v12.16.0
+    - v13.6.0
+    pr-url: https://github.com/nodejs/node/pull/31002
+    description: Remove the `--experimental-resolve-self` option.
+-->
+
+* Type: {string}
+
+```json
+{
+  "name": "package-name"
+}
+```
+
+The `"name"` field defines your package’s name. Publishing to the
+_npm_ registry may require a name that satisfies
+[certain requirements](https://docs.npmjs.com/files/package.json#name).
+
+The `"name"` field can be used in addition to the [`"exports"`][] field to
+[self-reference][] a package using its name.
+
+### `"type"`
+<!-- YAML
+added: v12.0.0
+changes:
+  - version:
+    - v12.17.0
+    - v13.2.0
+    pr-url: https://github.com/nodejs/node/pull/29866
+    description: Unflag `--experimental-modules`.
+-->
+
+* Type: {string}
+
+The `"type"` field defines the module format that Node.js will use for all
+`.js` files within a particular `package.json` file’s [package scope][].
+
+Files ending with `.js` will be loaded as ES modules when the nearest parent
+`package.json` file contains a top-level field `"type"` with a value of
+`"module"`.
+
+The nearest parent `package.json` is defined as the first `package.json` found
+when searching in the current folder, that folder’s parent, and so on up
+until a node_modules folder or the volume root is reached.
+
+```json
+// package.json
+{
+  "type": "module"
+}
+```
+
+```bash
+# In same folder as preceding package.json
+node my-app.js # Runs as ES module
+```
+
+If the nearest parent `package.json` lacks a `"type"` field, or contains	
+`"type": "commonjs"`, `.js` files are treated as [CommonJS][]. If the volume	
+root is reached and no `package.json` is found, Node.js defers to the default, a	
+`package.json` with no `"type"` field.
```

Comment:
```suggestion
`package.json` with no `"type"` field and `.js` files are treated as [CommonJS][].
```

@guybedford would that work for you? If feels a bit repetitive but at least it's explicit.

Reply:
Can we again just say "the default treatment as CommonJS" instead of explicitly a package.json with no type field?